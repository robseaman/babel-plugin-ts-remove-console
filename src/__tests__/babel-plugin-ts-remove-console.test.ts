import path from 'path';
import fs from 'fs';
import * as babel from '@babel/core';
import transformChangeIdentifier from '..';

const rootPath = (fixtureName: string): string =>
  path.join(path.resolve(), 'src', '__tests__', '__fixtures__', fixtureName);
const inputPath = (fixtureName: string): string =>
  path.join(rootPath(fixtureName), 'input.js');
const optionsPath = (fixtureName: string): string =>
  path.join(rootPath(fixtureName), 'options.js');
const readOptions = (fixtureName: string): string | undefined => {
  let optionsString;
  try {
    optionsString = fs.readFileSync(optionsPath(fixtureName), 'utf8');
  } catch (err) {
    // no options are passed as undefined, {} would work
    if (err.code === 'ENOENT') {
      return undefined;
    }
    throw err; // any other readfile error is a concern
  }
  let options;
  return eval(`options = ${optionsString}`); // allow eval to throw it's own errors
};

const runTransform = (
  transformChangeIdentifier: babel.PluginTarget,
  fixtureName: string,
): string => {
  const options = readOptions(fixtureName);

  const result = babel.transformFileSync(inputPath(fixtureName), {
    plugins: [[transformChangeIdentifier, options]],
    filename: fixtureName, // name for errors
    root: rootPath(fixtureName), // allow the fixture to contain babelrc
    configFile: false, // don't search
  });
  return result && result.code ? result.code : '';
};

describe('babel-plugin-ts-remove-console', () => {
  [
    'basic',
    'bound-excludes',
    'excludes',
    'guards',
    'local-binding',
    'member-expr-assignment-no-op',
    'member-expr-no-op',
    'replace-with-empty-block',
    'top-level',
    'top-level-stmts',
  ].forEach((testCase) => {
    it(`handles ${testCase}`, () => {
      expect(
        runTransform(transformChangeIdentifier, testCase),
      ).toMatchSnapshot();
    });
  });
});
