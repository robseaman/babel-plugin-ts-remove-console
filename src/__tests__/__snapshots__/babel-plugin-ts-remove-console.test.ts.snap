// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`babel-plugin-ts-remove-console handles basic 1`] = `
"function foo() {
  blah();
}"
`;

exports[`babel-plugin-ts-remove-console handles bound-excludes 1`] = `
"function foo() {
  const a = function () {};

  a();
  const b = console.error.bind(console);
  b(\\"asdf\\");
  blah();
}"
`;

exports[`babel-plugin-ts-remove-console handles excludes 1`] = `
"function foo() {
  console.error(\\"bar\\");
  blah();
  console.info(\\"blah\\");
}"
`;

exports[`babel-plugin-ts-remove-console handles guards 1`] = `
"function foo() {
  true && void 0;
  blah();
}"
`;

exports[`babel-plugin-ts-remove-console handles local-binding 1`] = `
"function foo(console) {
  console.foo(\\"hi\\");
  const bar = console.foo.bind(console);
}

function bar(a) {
  const {
    console
  } = a;

  a.b = console => console.bar(\\"bar\\");

  if (console.foo.call(console, \\"bar\\")) {
    return;
  }
}"
`;

exports[`babel-plugin-ts-remove-console handles member-expr-assignment-no-op 1`] = `
"function foo() {
  console.foo = function () {};

  console.error = function () {};
}"
`;

exports[`babel-plugin-ts-remove-console handles member-expr-no-op 1`] = `
"const a = function () {};

a();

const b = function () {};

b(\\"asdf\\");
var x = function () {} ? void 0 : foo();

function foo() {
  if (function () {}) {}
}"
`;

exports[`babel-plugin-ts-remove-console handles replace-with-empty-block 1`] = `
"if (blah) {}

for (;;) {}

for (var blah in []) {}

for (var blah of []) {}

while (blah) {}

do {} while (blah);"
`;

exports[`babel-plugin-ts-remove-console handles top-level 1`] = `
"true && void 0;
blah();"
`;

exports[`babel-plugin-ts-remove-console handles top-level-stmts 1`] = `"blah();"`;
